exporter/elasticsearchexporter/exporter_test.go

```diff
@@ -734,6 +734,36 @@ func TestExporterLogs(t *testing.T) {
 		assert.JSONEq(t, `{"a":"a","a.b":"a.b"}`, gjson.GetBytes(doc, `scope.attributes`).Raw)
 		assert.JSONEq(t, `{"a":"a","a.b":"a.b"}`, gjson.GetBytes(doc, `resource.attributes`).Raw)
 	})
+
+	t.Run("otel mode attribute complex value", func(t *testing.T) {
+		rec := newBulkRecorder()
+		server := newESTestServer(t, func(docs []itemRequest) ([]itemResponse, error) {
+			rec.Record(docs)
+			return itemsAllOK(docs)
+		})
+
+		exporter := newTestLogsExporter(t, server.URL, func(cfg *Config) {
+			cfg.Mapping.Mode = "otel"
+		})
+
+		logs := plog.NewLogs()
+		resourceLog := logs.ResourceLogs().AppendEmpty()
+		resourceLog.Resource().Attributes().PutEmptyMap("some.resource.attribute").PutEmptyMap("foo").PutStr("bar", "baz")
```

Comment:
While this kinda works, there are some limitations due to the way obj model gets serialized and de-dotted in OTel mode. The test would fail if, for example changing to 
```
resourceLog.Resource().Attributes().PutEmptyMap("some.resource.attribute").PutEmptyMap("foo.bar").PutStr("bar", "baz")
```

I think we should merge the change despite this limitation but think about longer-term alternatives.

I'm currently doing a POC of serializing pdata events directly to JSON for the OTel mode, without first converting  them do an `objmodel.Document`. This seems to work out quite well so far and I'd also suspect it to yield performance improvements due to one less layer of abstraction and allocation of temporary objects. WDYT of that approach?

Reply:
Here's the draft PR for that POC: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37032